openapi-model/src/main/scala/sttp/apispec/openapi/validation/OpenAPICompatibilityIssue.scala

```diff
@@ -0,0 +1,152 @@
+package sttp.apispec.openapi.validation
+
+import sttp.apispec.openapi.ResponsesKey
+import sttp.apispec.validation.SchemaCompatibilityIssue
+
+sealed abstract class OpenAPICompatibilityIssue {
+  def description: String
+
+  override def toString: String = description
+  protected def issuesRepr(issues: List[OpenAPICompatibilityIssue]): String =
+    issues.iterator
+      .map(i => s"- ${i.description.replace("\n", "\n  ")}") // indent
+      .mkString("\n")
+}
+
+sealed abstract class OpenAPICompatibilitySubIssues extends OpenAPICompatibilityIssue {
+  def subIssues: List[OpenAPICompatibilityIssue]
+}
+
+case class MissingPath(pathName: String) extends OpenAPICompatibilityIssue {
+  def description: String =
+    s"missing path: $pathName"
+}
+
+case class IncompatiblePath(
+    pathName: String,
+    subIssues: List[OpenAPICompatibilityIssue]
+) extends OpenAPICompatibilitySubIssues {
+  def description: String =
+    s"incompatible path $pathName:\n${issuesRepr(subIssues)}"
+}
+
+case class MissingOperation(httpMethod: String) extends OpenAPICompatibilityIssue {
+  def description: String =
+    s"missing operation for $httpMethod method"
+}
+
+case class IncompatibleOperation(
+    httpMethod: String,
+    subIssues: List[OpenAPICompatibilityIssue]
+) extends OpenAPICompatibilitySubIssues {
+  def description: String =
+    s"incompatible operation $httpMethod:\n${issuesRepr(subIssues)}"
+}
+
+case class MissingParameter(
+    name: String
+) extends OpenAPICompatibilityIssue {
+  def description: String =
+    s"missing parameter $name"
+}
+
+case class IncompatibleRequiredParameter(
+    name: String
+) extends OpenAPICompatibilityIssue {
+  def description: String =
+    s"parameter '$name' is required by the client but optional on the server"
+}
+
+case class IncompatibleParameter(
+    name: String,
+    subIssues: List[OpenAPICompatibilityIssue]
+) extends OpenAPICompatibilitySubIssues {
```

Comment:
hm so here I'm wondering if the hierarchy is correct - we've got `Issues` and `SubIssues`, which are also `Issues`. Can `IncompatibleParameter` be used as a top-level issue, or only as a sub-issue of `IncompatibleRequiredParameter`? Should `SubIssue extend Issue`?

Reply:
so `IncompatibleRequiredValue` can be used as a top-level issue as well? That is, it can be used in two contexts: as a top-level and sub-level issue?